lib/recursive_open_struct.rb

```diff
@@ -92,4 +105,25 @@ def display_recursive_open_struct(io, ostrct_or_hash, indent_level, recursion_li
     true
   end
 
+  private
+  
+  def deep_dup(obj, visited=[])
+    if obj.is_a?(Hash)
+      obj.each_with_object({}) do |(key, value), h|
+        h[key] = value_or_deep_dup(value, visited)
+      end
+    elsif obj.is_a?(Array)
```

Comment:
Hmm, what if `recurse_over_arrays` is false? I think a good case can be made that when `recurse_over_arrays` is false, that the `deep_dup` should stop with the original hash tree --- it shouldn't `dup` or recurse into any Arrays it encounters.